### PR TITLE
Rabin karp in c

### PR DESCRIPTION
**Pull request by disha2sinha**
Fixes #2161

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- 
- 

#### Files Added:
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Rabin_Karp_in_C/Rabin-Karp.c [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#include <stdio.h>
```
Comment (jainaman224) [inline Rabin_Karp_in_C/Rabin-Karp.c]: Fix file name to `Rabin_Karp`.
Fix Folder name to `Rabin_Karp`

Like https://github.com/jainaman224/Algo_Ds_Notes/tree/master/Rabin_Karp

File: Rabin_Karp_in_C/Rabin-Karp.c [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+#include<string.h>
+
+void Rabin_Karp(char *text, char *pattern)
+{
+    int text_len = strlen(text);
+    int pattern_len = strlen(pattern);
+    int hash=1;
+    
+    for(int i=0;i<pattern_len-1;i++)
```
Comment (jainaman224) [inline Rabin_Karp_in_C/Rabin-Karp.c]: Add space around operators like `=`, `==`, `<`, `+`, `-`.

Comment (jainaman224) [inline Rabin_Karp_in_C/Rabin-Karp.c]: Add space before operators like `(`, `{`.

File: Rabin_Karp_in_C/Rabin-Karp.c [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    {
+        hash=(hash*256)%149;
+        //there are 256 total characters in Ascii table & 149 is a prime number chosen randomly
+    }
+    
+    int hash_txt=0,hash_pat=0;
+    
+    for(int i=0;i<pattern_len;i++)
```
Comment (jainaman224) [inline Rabin_Karp_in_C/Rabin-Karp.c]: Add space after operator like `;`, `,`, `//`.

File: Rabin_Karp_in_C/Rabin-Karp.c [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            {
+                hash_txt=hash_txt+149;
+            }
+            
+        }
+        
+    }
+
+    
+
```
Comment (jainaman224) [inline Rabin_Karp_in_C/Rabin-Karp.c]: Remove unnecessary new line.

**Comment by disha2sinha** [comment]
Changes made & a new PR #2389 is opened.

**Pull request closed without merging**